packages/server-wallet/src/models/objective.ts

```diff
@@ -31,43 +30,32 @@ function extractReferencedChannels(objective: Objective): string[] {
 
 type ObjectiveStatus = 'pending' | 'approved' | 'rejected' | 'failed' | 'succeeded';
 
-/**
- * Objectives that are currently supported by the server wallet (wire format)
- */
-export type SupportedWireObjective = OpenChannel | CloseChannel;
-
-export type DBOpenChannelObjective = OpenChannel & {objectiveId: string; status: ObjectiveStatus};
-export type DBCloseChannelObjective = CloseChannel & {
-  objectiveId: string;
-  status: ObjectiveStatus;
-};
-export type DBDefundChannelObjective = {
-  type: 'DefundChannel';
-  participants: Participant[];
+type WalletObjective<O extends Objective> = O & {
   objectiveId: string;
   status: ObjectiveStatus;
-  data: {targetChannelId: string};
+  createdAt: string;
+  progressLastMadeAt: string;
 };
 
-export type DBSubmitChallengeObjective = {
-  data: {targetChannelId: string};
-  participants: Participant[];
-  objectiveId: string;
-  status: ObjectiveStatus;
-  type: 'SubmitChallenge';
-};
+export type DBOpenChannelObjective = WalletObjective<OpenChannel>;
```

Comment:
Nit: this change should have probably been broken up. Otherwise, the functional change (the addition of the timestamp) is mixed in with the refactor, which shouldn't change behaviour.